src/queues.c

```diff
@@ -331,25 +333,34 @@ void queues_update(void)
         }
 
         /* move notifications from queue to displayed */
-        while (waiting->length > 0) {
+        GList *iter = g_queue_peek_head_link(waiting);
+        while (iter) {
+                notification *n = iter->data;
+                GList *nextiter = iter->next;
```

Comment:
[nitpick] Is there a reason for having `nextiter` var rather than directly referring to `iter->next`?

Reply:
Must be an older relict of the times, where I wanted to replace all queue iteration things with two pointers and a for loop.